### PR TITLE
Implement Languages as a Mapping and Iterable type

### LINKED ISSUE #5657: Languages incorrectly claims it is a tuple
It should be a mapping, but the bug is it claims to be a tuple.
Replace with 'object'.

### LINKED ISSUE #5658: Languages should be a Mapping
#5657 first. Then make Languages a Mapping and Iterable class (see collections)

### PR DESCRIPTION
**Pull request by corona10**
Implement Languages as a Mapping and Iterable type by using
a tuple as the instance variable.

~~Closes #5657~~ 
Closes #5658

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by mach64** [comment]
You have to get assigned to an issue before you can start working on it. Please read and follow the instructions in [Newcomer's Guide](https://coala.io/newcomer) first.

**Comment by corona10** [comment]
@mach64 I know about the process but as you know corobo assign doesn't works now day. And this issue was not processed last 3 month

**Comment by mach64** [comment]
@corona10 If you joined the gitter channels, you would notice that even though corobo is down, people wait for maintainers to manually assign the issue to them. This is what should have been done here.

File: tests/bearlib/languages/LanguageTest.py [hunk 1]
```diff
@@ -46,3 +48,62 @@ def test_invalid_attribute(self):
...
+        self.assertIn('Python', self.langs[1])
+        reverse = list(reversed(self.langs))
+        self.assertIn('CPP', reverse[1])
+        self.assertIn('Python', reverse[0])
+        self.assertEqual(self.langs, self.expected_langs)
+        langs2 = Languages(self.expected_langs)
+        self.assertTrue(self.langs == langs2)
+        self.assertFalse(self.langs != langs2)
+        self.assertFalse(self.langs == [])
+        cnt = 0
```
Comment (jayvdb) [inline tests/bearlib/languages/LanguageTest.py]: create a separate test method for this chunk of code, as it is more procedural and detailed, and needs to be read more carefully.

File: tests/bearlib/languages/LanguageTest.py [hunk 2]
```diff
@@ -46,3 +48,62 @@ def test_invalid_attribute(self):
...
+        while True:
+            try:
+                next(it)
+                it_cnt += 1
+            except StopIteration:
+                break
+        self.assertEqual(cnt, 2)
+        self.assertEqual(it_cnt, 2)
+        self.assertEqual(str(self.langs),
+                         '(CPP, Python 2.7, 3.3, 3.4, 3.5, 3.6)')
```
Comment (jayvdb) [inline tests/bearlib/languages/LanguageTest.py]: this isnt very useful with commas in unpredictable locations.

File: tests/bearlib/languages/LanguageTest.py [hunk 3]
```diff
@@ -46,3 +48,62 @@ def test_invalid_attribute(self):
...
+
+
+class LanguagesCollectionDataTypeTest(unittest.TestCase):
+
+    def setUp(self):
+        self.expected_langs = (Language['CPP'], Language['Python'])
+        self.langs = Languages(self.expected_langs)
+
+    def tearDown(self):
+        pass
```
Comment (jayvdb) [inline tests/bearlib/languages/LanguageTest.py]: unnecessary method

File: tests/bearlib/languages/LanguageTest.py [hunk 4]
```diff
@@ -46,3 +48,62 @@ def test_invalid_attribute(self):
     def test_attribute_list_empy(self):
         with self.assertRaisesRegex(AttributeError, 'no available attribute'):
             self.lang_unknown.not_an_attribute
+
+
+class LanguagesCollectionDataTypeTest(unittest.TestCase):
+
+    def setUp(self):
+        self.expected_langs = (Language['CPP'], Language['Python'])
+        self.langs = Languages(self.expected_langs)
```
Comment (jayvdb) [inline tests/bearlib/languages/LanguageTest.py]: `self.uut = ...`

**Pull request closed without merging**